### ISSUE TITLE
Add better vim-devicons support (suggested options)

### ISSUE THREAD

**Issue by ryanoasis** [ctrlpvim/ctrlp.vim#267]
## Overview

This would solve 1 problem ([MRU current buffer in vim-devicons](https://github.com/ryanoasis/vim-devicons/issues/88)) and 1 feature/improvement (add glyphs to other ctrlp modes besides MRU).

So I submitted a PR a while back (#104) to allow for customizing display of MRU.

However there was a problem (https://github.com/ryanoasis/vim-devicons/issues/88) with adding the glyphs at the 'reformat' stage because it breaks the logic of 'MatchIt' where the current open buffer is 'excluded' from the list (lines).

So now I want to submit another PR but I see 2 different ways to accomplish it:

### option 1: 
Configuration that can be used in other cases and not just for vim-devicons:

https://gist.github.com/ryanoasis/6705561d4f4ab6143ef00b04fe11ed84

### option 2:
Direct support for vim-devicons added directly into ctrlp:

https://gist.github.com/ryanoasis/119df97df148214ae4f2bee91239b127

this would be very similar to how vim-devicons was integrated with vim-startify: https://github.com/mhinz/vim-startify/commit/c193556225af93405cc87518781d48cf932efc2f

--

**TL;DR:** Essentially this would help me accomplish 1 bug fix and 1 feature with fairly simple logic addition

Any opinions on this? Specifically if option 1 or option 2 preferred or if there is another way someone sees :smile:

**Comment by liuxyc** [comment]
bump this :+1:

**Comment by mattn** [comment]
I prefer option1 but please wait until I will be able to take a enough time to try this.

**Comment by ryanoasis** [comment]
@mattn Sure I understand. Thank you

**Comment by mattn** [comment]
merged https://github.com/ctrlpvim/ctrlp.vim/commit/5aeab2af35b2175b787cd7b8f3f1fbb7f9b631c2

**Comment by ryanoasis** [comment]
@mattn thanks! 👍 😄
